homeassistant/components/roborock/scene.py

```diff
@@ -0,0 +1,86 @@
+"""Support for Roborock scene."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Coroutine
+from typing import Any
+
+from roborock.containers import HomeDataScene
+
+from homeassistant.components.scene import Scene as SceneEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RoborockConfigEntry
+from .coordinator import RoborockDataUpdateCoordinator, RoborockDataUpdateCoordinatorA01
+from .entity import RoborockEntity
+
+
+async def _setup_coordinator(
+    coordinator: RoborockDataUpdateCoordinator | RoborockDataUpdateCoordinatorA01,
+) -> list[SceneEntity]:
+    scenes = await coordinator.rest_api.get_scenes()
+    return [RoborockSceneEntity(coordinator, scene) for scene in scenes]
+
+
+def _build_setup_functions(
+    coordinators: list[
+        RoborockDataUpdateCoordinator | RoborockDataUpdateCoordinatorA01
+    ],
+) -> list[
+    Coroutine[
+        Any,
+        Any,
+        list[SceneEntity],
+    ]
+]:
+    """Create a list of setup functions that can later be called asynchronously."""
+    return [_setup_coordinator(coordinator) for coordinator in coordinators]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: RoborockConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up scene platform."""
+    coordinator_entities = await asyncio.gather(
+        *_build_setup_functions(config_entry.runtime_data.values()),
```

Comment:
This 3 layers of nesting of function calls seems like it should just all be inlined here.

Reply:
Did my best to inline them, but Im not a strong Python coder and getting an inline coroutine was a failed attempt on my side, Please let me know how I can do it even better if possible